components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/internal/dao/impl/ActionManagementDAOFacade.java

```diff
@@ -304,6 +304,8 @@ private void addEncryptedAuthSecretsToBuilder(ActionDTOBuilder actionDTOBuilder,
                                 .type(actionDTOBuilder.getEndpoint().getAuthentication().getType())
                                 .properties(encryptedPropertyMap)
                                 .build())
+                        .allowedHeaders(actionDTOBuilder.getEndpoint().getAllowedHeaders())
+                        .allowedParameters(actionDTOBuilder.getEndpoint().getAllowedParameters())
```

Comment:
These headers and params are nullable based on your implementation above. Just noting down